src/main/java/seedu/address/logic/commands/EditCommand.java

```diff
@@ -101,6 +106,23 @@ public CommandResult execute(Model model) throws CommandException {
             throw new CommandException(MESSAGE_DUPLICATE_PERSON);
         }
 
+        if (this.editPersonDescriptor.isRoleEdited()) {
+            if (!isNull(personToEdit.getAffiliations()) && personToEdit.getAffiliations().isEmpty()) {
+                throw new CommandException(MESSAGE_EDIT_ROLE_CONTAIN_AFFILIATION);
+            }
+        }
+
+        if (this.editPersonDescriptor.isNameEdited()) {
+            AffiliationModifier.nameChangeAffiliations(personToEdit.getAffiliations(), personToEdit.getName(),
+                    editedPerson.getName(), model);
+        }
+
+        if (this.editPersonDescriptor.isAffiliationEdited() && !editedPerson.getAffiliations().isEmpty()) {
```

Comment:
There may be a logical issue here. Maybe we can remove the check on whether the affiliation set is empty? Since having an empty set implies the clearing of the set, while being null implies not edited.

Reply:
Thanks for pointing it out. Will change it in the next commit.